source/basic.tex

```diff
@@ -3352,7 +3352,8 @@
 except that if the object is a union member or subobject thereof,
 its lifetime only begins if that union member is the
 initialized member in the union\iref{dcl.init.aggr,class.base.init},
-or as described in \ref{class.union} and \ref{class.copy.ctor},
+or as described in
+\ref{class.union}, \ref{class.copy.ctor}, and \ref{class.copy.assign},
```

Comment:
Of these cases, only [class.copy.ctor] (calling a union copy constructor) starts the lifetime of the union member subobject. Both the [class.union] (assigning to a union member) and [class.copy.assign] (assigning to a union) cases create a new object that transparently replaces the union member. So I think only [class.copy.ctor] is in scope here, although the reason for that is subtle.

Reply:
I suppose another viewpoint is that, because the newly-created object transparently replaces a union member subobject, it is itself a member subobject ([intro.object]p2), and so it is in the scope of this rule, in which case this change seems appropriate. Either way, we should be consistent: either both [class.union] and [class.copy.assign] should be listed here or neither of them.